samples/net/openthread/coap_server/src/ot_coap_utils.c

```diff
@@ -0,0 +1,247 @@
+#include <logging/log.h>
+LOG_MODULE_REGISTER(ot_coap_utils, CONFIG_OT_COAP_UTILS_LOG_LEVEL);
+
+#include "ot_coap_utils.h"
+
+#include <net/net_pkt.h>
+#include <net/net_l2.h>
+#include <net/openthread.h>
+#include <openthread/message.h>
+#include <openthread/ip6.h>
+#include <openthread/coap.h>
+#include <openthread/thread.h>
+
+typedef struct {
```

Comment:
doesn't it generate a warning? Not sure if its reported by compile or checkpatch but if I recall correctly typedefs are not allowed in c files.

Reply:
I don't see any warnings